README.md

```diff
@@ -1 +1,122 @@
+    return nothing
+end
+```
+
+Now that we defined all overloads and contexts we can define our user-facing function:
+
+```julia
+# Prints relative positions of `\n` in each ingested chunk of size `buffersize`,
+# using `nworkers` spawned tasks.
+function print_newlines(io, buffersize, nworkers)
+    lexer = Lexer(io, nothing, '\n')
+    parsing_ctx = ParsingContext()
+    consume_ctx = ConsumeContext()
+    chunking_ctx = ChunkingContext(buffersize, nworkers, 0, nothing)
+    # ChunkedBase.jl requires 2 result buffers per worker task, we'd get an error otherwise
+    result_buffers = [ResultBuffer(Int[]) for _ in 1:2nworkers]
+    # We need to fill the byte buffer and find the newlines in it before we hand
+    # the `consume_ctx` to the function
+    ChunkedBase.read_and_lex!(lexer, chunking_ctx)
+    parse_file_parallel(lexer, parsing_ctx, consume_ctx, chunking_ctx, result_buffers)
+    return nothing
+end
+```
+Let's run in on some data:
+```julia
+julia> io = IOBuffer((("x" ^ 4095) * "\n") ^ 64); # 256KiB
+julia> print_newlines(io, 64*1024, 4);
+# [ Info: Newlines in chunk (id:(1, 1)): [16384, 20480, 24576, 28672, 32768, 36864]
+# [ Info: Newlines in chunk (id:(1, 1)): [0, 4096, 8192, 12288, 16384]
+# [ Info: Newlines in chunk (id:(1, 1)): [36864, 40960, 45056, 49152, 53248, 57344]
+# [ Info: Newlines in chunk (id:(1, 1)): [57344, 61440, 65536]
+# [ Info: Newlines in chunk (id:(2, 1)): [16384, 20480, 24576, 28672, 32768, 36864]
+# [ Info: Newlines in chunk (id:(2, 1)): [0, 4096, 8192, 12288, 16384]
+# [ Info: Newlines in chunk (id:(2, 1)): [36864, 40960, 45056, 49152, 53248, 57344]
+# [ Info: Newlines in chunk (id:(2, 1)): [57344, 61440, 65536]
+# [ Info: Newlines in chunk (id:(1, 2)): [0, 4096, 8192, 12288, 16384]
+# [ Info: Newlines in chunk (id:(1, 2)): [36864, 40960, 45056, 49152, 53248, 57344]
+# [ Info: Newlines in chunk (id:(1, 2)): [16384, 20480, 24576, 28672, 32768, 36864]
+# [ Info: Newlines in chunk (id:(1, 2)): [57344, 61440, 65536]
+# [ Info: Newlines in chunk (id:(2, 2)): [0, 4096, 8192, 12288, 16384]
+# [ Info: Newlines in chunk (id:(2, 2)): [16384, 20480, 24576, 28672, 32768, 36864]
+# [ Info: Newlines in chunk (id:(2, 2)): [36864, 40960, 45056, 49152, 53248, 57344]
+# [ Info: Newlines in chunk (id:(2, 2)): [57344, 61440, 65536]
+```
+Behind the scenes, `ChunkedBase.jl` was using two 64KiB buffers, finding newlines in them, and splitting the found newlines among 4 tasks. We can see that each of the buffers (identified by the first number in the `id` tuple) was refilled two times (the refill number is the second element of the tuple).
+The way we set up our data, there should be one newline every 4KiB of input, so we'd expect 16 newlines per chunk, but we could see that there are 20 numbers reported per chunk -- this is because the first element of the newline segment we send to the tasks is either 0 or the end of the previous sub-chunk, so we get 4 duplicated elements.
```

Comment:
I also don't exactly understand this. Are you saying this is a gotcha that users of ChunkedBase need to understand? Or is this an artifact of how you set up this example? I don't understand where the 0s are coming from here.

Reply:
I've reworded this part, hopefully, its better now.